src/store/singing.ts

```diff
@@ -160,11 +160,23 @@ type Singer = {
 };
 
 type Phrase = {
-  readonly hash: string;
   readonly singer: Singer | undefined;
-  readonly notes: Note[];
-  readonly noteEvents: NoteEvent[];
-  readonly audioEvents: AudioEvent[];
+  readonly score: Score;
+  query?: AudioQuery;
+  queryHash?: string; // queryの変更を検知するためのハッシュ
+  buffer?: AudioBuffer;
+  startTime?: number;
```

Comment:
renderingが進むにつれてここのデータが代入されていく、という設計なんですね･･･！
なぜ全部optionalなのか難しいので、メモ残すとわかりやすいかもと思いました。
```suggestion
  // renderingが進むに連れてデータが代入されていく
  query?: AudioQuery;
  queryHash?: string; // queryの変更を検知するためのハッシュ
  buffer?: AudioBuffer;
  startTime?: number;
```

Reply:
こちらコメントを追加しました！